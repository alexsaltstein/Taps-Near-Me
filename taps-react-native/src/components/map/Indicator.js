import React from 'react';
import {TouchableOpacity, View, StyleSheet} from 'react-native';
import { COLORS } from '../../styles/COLORS';

const Indicator = ({ width, onPress }) => {
  return (
    <TouchableOpacity
      onPress={() => onPress()}
      activeOpacity={1}>
      <View style={{
        width: width,
        height: width,
        borderRadius: width / 2,
        backgroundColor: COLORS.white,
        margin: 5
      }}
      />
    </TouchableOpacity>
  )
}

export default Indicator;