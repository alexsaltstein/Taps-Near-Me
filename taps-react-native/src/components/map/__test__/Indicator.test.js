import '../../../matchMedia.mock';
import React from 'react';
import renderer, { act } from 'react-test-renderer';
import Indicator from '../Indicator';
import { RecoilRoot } from 'recoil';
import { render, fireEvent } from '@testing-library/react-native'
import { View, Text, StyleSheet, TouchableOpacity, TextInput } from 'react-native';
import ReactDOM from 'react-dom';

test('renders correctly', () => {
  const tree = renderer.create(<RecoilRoot><Indicator width= {5} onPress= {() => onPress()}/></RecoilRoot>).toJSON();
  expect(tree).toMatchSnapshot();
});