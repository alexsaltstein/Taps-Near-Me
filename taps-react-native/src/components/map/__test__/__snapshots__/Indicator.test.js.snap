// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
  data-focusable={true}
  onClick={[Function]}
  onContextMenu={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "WebkitTransitionDuration": "0s",
      "transitionDuration": "0s",
    }
  }
  tabIndex="0"
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,1.00)",
        "borderBottomLeftRadius": "2.5px",
        "borderBottomRightRadius": "2.5px",
        "borderTopLeftRadius": "2.5px",
        "borderTopRightRadius": "2.5px",
        "height": "5px",
        "marginBottom": "5px",
        "marginLeft": "5px",
        "marginRight": "5px",
        "marginTop": "5px",
        "width": "5px",
      }
    }
  />
</div>
`;
